pkg/virt-handler/container-disk/mount.go

```diff
@@ -222,6 +222,12 @@ func (m *mounter) setAddMountTargetRecordHelper(vmi *v1.VirtualMachineInstance,
 func (m *mounter) MountAndVerify(vmi *v1.VirtualMachineInstance) (map[string]*containerdisk.DiskInfo, error) {
 	record := vmiMountTargetRecord{}
 	disksInfo := map[string]*containerdisk.DiskInfo{}
+	kb := vmi.Spec.Domain.Firmware.KernelBoot.Container
+
+	if kb.KernelPath == "" {
+		return nil, fmt.Errorf(
+			"Invalid kernel or initrd in image '%s'.", kb.Image)
+	}
```

Comment:
@aliamerj thank you for this contribution!

Here are some problems I see with this code:
1. In this specific place, `vmi.Spec.Domain.Firmware.KernelBoot` might be nil, and for good reasons. Not only VM that reaches this code uses an external kernel, but could use a container disk. Probably the more suiting place would be under `mountKernelArtifacts()`, right after `HasKernelBootContainerImage()` is being called.
2. This code is not tested. Please make sure to add some unit tests.

Thank you!

Reply:
@iholder101
how about make it like the `HasKernelBootContainerImage`
for example: 
```go
	if !util.HasValidKernelBootContainerImage(vmi) {
		log.Log.Object(vmi).Infof("Invalid kernel boot in image '%s'.", kb.Image)
		return nil
	}
```

Reply:
@aliamerj It generally makes sense. But now on a second look, there are already checks for whether the path exists there:
https://github.com/kubevirt/kubevirt/blob/e307800b8fc56d83a82c461572bbb82fdab78187/pkg/virt-handler/container-disk/mount.go#L457

Perhaps you can check if one of the paths exist in `HasKernelBootContainerImage()`. If both exists we should exit early anyhow. If one exist, we should know how to only mount it and ignore the other. In addition I think that if the kernelboot is defined without both kernel and initrd paths, this should result in an error and not just a log entry.

Hope this makes sense.

Reply:
if we put both the logic in the `HasKernelBootContainerImage()`
maybe we should make it like this ??
```go 
func HasKernelBootContainerImage(vmi *v1.VirtualMachineInstance) error {
	if vmi == nil {
		return fmt.Errorf("VirtualMachineInstance is nil")
	}

	vmiFirmware := vmi.Spec.Domain.Firmware
	if vmiFirmware == nil || vmiFirmware.KernelBoot == nil || vmiFirmware.KernelBoot.Container == nil {
		return fmt.Errorf("kernel boot is not defined")
	}

	kb := vmiFirmware.KernelBoot.Container
	kernelExists := kb.KernelPath != ""
	initrdExists := kb.InitrdPath != ""

	if kernelExists && initrdExists {
		return nil // Both paths are valid
	} else if kernelExists || initrdExists {
		return nil // Only one path is valid
	}

	return fmt.Errorf("both kernel and initrd paths are not defined in image '%s'", kb.Image)
}

```

Reply:
@iholder101 
and we can remove  the `HasKernelBootContainerImage(vmi)` from `util` package 
**If vmi and vmiFirmware are required to have**  
what do you think ?